src/sdk/data-api.ts

```diff
@@ -86,15 +86,25 @@ export class DataApiImpl implements DataApi {
 
   async query(soql: string): Promise<RecordQueryResult> {
     return this.promisifyRequests(async (conn: Connection) => {
-      const response = await conn.query(soql);
-      const records = response.records.map(createCaseInsensitiveRecord);
-
-      return {
-        done: response.done,
-        totalSize: response.totalSize,
-        records,
-        nextRecordsUrl: response.nextRecordsUrl,
-      };
+      try {
+        const response = await conn.query(soql);
+        const records = response.records.map(createCaseInsensitiveRecord);
+        return {
+          done: response.done,
+          totalSize: response.totalSize,
+          records,
+          nextRecordsUrl: response.nextRecordsUrl,
+        };
+      } catch (e) {
+        if (e.errorCode == "ERROR_HTTP_404") {
+          try {
+            JSON.parse(e.message);
+          } catch (e) {
+            throw new Error("Could not parse API response as JSON!");
```

Comment:
It looks like when there's an error, and it's not parsable as JSON that the contents of the message are going to be hidden. Should we put the error contents in the error message? When that happens will the developer have enough context to figure out the problem?

Reply:
@schneems The contents of this error has an error code equal to `ERROR_HTTP_404` and error message is the entire body of the api response which isn't super helpful for developers either: 
```
<!doctype html>
<html>
<head>
    <title>Example Domain</title>

    <meta charset="utf-8" />
    <meta http-equiv="Content-type" content="text/html; charset=utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type="text/css">
    body {
        background-color: #f0f0f2;
        margin: 0;
        padding: 0;
        font-family: -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
        
    }
    div {
        width: 600px;
        margin: 5em auto;
        padding: 2em;
        background-color: #fdfdff;
        border-radius: 0.5em;
        box-shadow: 2px 3px 7px 2px rgba(0,0,0,0.02);
    }
    a:link, a:visited {
        color: #38488f;
        text-decoration: none;
    }
    @media (max-width: 700px) {
        div {
            margin: 0 auto;
            width: auto;
        }
    }
    </style>    
</head>

<body>
<div>
    <h1>Example Domain</h1>
    <p>This domain is for use in illustrative examples in documents. You may use this
    domain in literature without prior coordination or asking for permission.</p>
    <p><a href="https://www.iana.org/domains/example">More information...</a></p>
</div>
</body>
</html>
```